packages/terra-popup/src/_TetherComponent.jsx

```diff
@@ -0,0 +1,251 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Tether from 'tether';
+
+const ATTACHMENT_POSITIONS = [
+  'top left',
+  'top center',
+  'top right',
+  'middle left',
+  'middle center',
+  'middle right',
+  'bottom left',
+  'bottom center',
+  'bottom right',
+];
+
+const propTypes = {
+  /**
+   * A hash of tether classes which should be changed or disabled.
+   */
+  classes: PropTypes.object,
+  /**
+   * The prefix placed at the beginning of the default classes.
+   */
+  classPrefix: PropTypes.string,
+  /**
+   * Rule set to pass to tether, contraining the content to by setting 'to', 'attachment', and/or 'pin'.
+   * Also allows for class aliasing or 'outOfBoundsClass' and 'pinnedClass'.
+   */
+  constraints: PropTypes.array,
+  /**
+   * The content to be tethered.
+   */
+  content: PropTypes.element,
+  /**
+   * String pair of top, middle, bottom, and left, center, right.
+   */
+  contentAttachment: PropTypes.oneOf(ATTACHMENT_POSITIONS).isRequired,
+  /**
+   * String pair of top and left offset, ie "10px -4px".
+   */
+  contentOffset: PropTypes.string,
+  /**
+   * Should tethering be disabled following the initial positioning.
+   */
+  disableOnPosition: PropTypes.bool,
+  /**
+   * Should element be tethered to the page.
+   */
+  isEnabled: PropTypes.bool,
+  /**
+   * The ability to disable "gpu" and "moveElement" optimizations.
```

Comment:
Does this need to be of type `PropType.shape` to help guide the user of the 'gpu' and 'moveElement' structures? Also, does this prop give the ability to enable certain optimizations as well? If no, possibly renaming prop to `disableOptimizations` might be clearer.